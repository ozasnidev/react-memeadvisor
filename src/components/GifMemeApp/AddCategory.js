import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [newCategory, setNewCategory] = useState('');
    const handleNewInputValue = (event) => {
        setNewCategory(event.target.value);
    }
    const handleSubmit = (event) => {
        event.preventDefault();
        if(newCategory.trim().length > 2){
            setCategories(categories => [newCategory, ...categories]);
            setNewCategory('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <div className="panel-block">
            <p className="control">
                <input className="input is-success" 
                    type="text" 
                    value={newCategory}
                    onChange={handleNewInputValue}
                    placeholder="Search" 
                />
            </p>
            </div>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
