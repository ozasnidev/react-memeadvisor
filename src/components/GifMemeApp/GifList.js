import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif';
import PropTypes from 'prop-types';

export const GifList = ({category}) => {

    const {data,loading} = useFetchGif( category );

    return (
        <div>
            <div className="panel-block">
                <h4 className="title is-4 animate__animated animate__fadeInDown"> {category} </h4>
                {loading && <div className="spinner"></div>}
            </div>
            <div className="img-thumbnail-container">
                {data.map(image => ( 
                    <img src={image.url} alt={image.title} key={image.id} className="img-thumbnail  animate__animated animate__fadeInUp"/>
                ))}
            </div>
        </div>
    )
}

GifList.propTypes = {
    category: PropTypes.string.isRequired
}