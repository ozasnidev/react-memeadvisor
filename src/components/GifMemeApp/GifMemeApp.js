import React, {useState} from 'react';
import NavbarMeme from '../Shared/navbar/Navbar';
import { AddCategory } from './AddCategory';
import { GifList } from './GifList';


const GifMemeApp = ({ defaultCategories = []}) => { 
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <NavbarMeme />
            <div className="container">
                <div className="add-category-container">
                    <article className="panel is-success">
                        <p className="nav-title panel-heading">Categories</p>
                        <AddCategory setCategories = {setCategories}/>
                        {categories.map(category => <GifList key={category} category={category} />)} 
                    </article>
                </div>
            </div>
        </>
    )   
}
export default GifMemeApp;