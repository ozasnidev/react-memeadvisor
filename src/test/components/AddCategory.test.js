import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/GifMemeApp/AddCategory';

describe('AddCategory component test', () => {
    const setCategories = jest.fn();
    const value = 'Goku';
    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } /> );
    });

    test('should to show AddCategory component without issues', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should to change the input text box', () => {
        wrapper.find('input').simulate('change', { target: { value } });
    });

    test('should not to post the info with submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should to call setCategories and reset the input', () => {
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault: () => {} });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value')).toBe('');
    });
});