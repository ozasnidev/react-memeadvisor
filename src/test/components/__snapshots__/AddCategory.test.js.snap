// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component test should to show AddCategory component without issues 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="panel-block"
  >
    <p
      className="control"
    >
      <input
        className="input is-success"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </p>
  </div>
</form>
`;
